content/en/developer-guide/development-environment-setup.md

```diff
@@ -0,0 +1,89 @@
+
+## Development requirements
+
+You will need the following tools for the local development of LocalStack.
+
+* [Python 3.10+](https://www.python.org/downloads/)
+* [Sasl](https://packages.debian.org/buster/libsasl2-2)
+* `pip`
+* [`virtualenv`](https://pypi.org/project/virtualenv/)
+* [OpenJDK](https://openjdk.org/install/)
+* [NodeJS & npm](https://nodejs.org/en/download/)
+* [Maven](https://maven.apache.org/download.cgi)
+* [Gradle](https://gradle.org/install/)
+* [Terraform](https://www.terraform.io/downloads)
+* [Docker](https://docs.docker.com/desktop/)
+* [Docker-Compose](https://docs.docker.com/compose/install/)
+
+We recommend you to individually install the above tools using your favorite package manager. For example, on macOS, you can use [Homebrew](https://brew.sh/) to install the above tools.
+
+### Setting up Development Environment
+
+To make contributions to LocalStack, you need to be able to run LocalStack in host mode from your IDE, and be able to attach a debugger to the running LocalStack instance. We have a basic tutorial to cover how you can do that.
+
+The basic steps include:
+
+*   Clone the localstack repository, and optionally our fork of `moto`:
+    -   [https://github.com/localstack/localstack/](https://github.com/localstack/localstack/)
+    -   [https://github.com/localstack/moto](https://github.com/localstack/moto)
+*   Make sure you have `javac`, `node`, `npm`, and Python 3.10 installed.
+*   Most of the contributors use the free community version [PyCharm](https://www.jetbrains.com/pycharm/).
+
+{{< youtube XHLBy6VKuCM >}}
+
+If you pull the repo in order to extend/modify LocalStack, run this command to install all the dependencies:
+
+{{< command >}}
+$ make install
+{{< /command >}}
+
+This will install the required pip dependencies in a local Python `virtualenv` directory `.venv` (your global python packages will remain untouched), as well as some node modules in `./localstack/node_modules/`. Depending on your system, some `pip`/`npm` modules may require additional native libs installed.
+
+The Makefile contains a start command to conveniently start:
+
+{{< command >}}
+$ make start
+{{< / command >}}
+
+### Host mode
+
+Although we strongly recommend to use Docker, the infrastructure can also be spun up directly on the host machine using the `--host` startup flag. Note that this will require additional dependencies, and is not supported on some operating systems, including Windows.
+
+{{< command >}}
+$ localstack start --host
+{{< / command >}}
+
+LocalStack will attempt to automatically fetch the missing dependencies when you first start it up in `host` mode; Alternatively, you can use the full profile to install all dependencies at `pip` installation time:
+
+{{< command >}}
+$ pip install "localstack[full]"
+{{< / command >}}
```

Comment:
this is redundant information, and somewhat incoherent, given we just set up the localstack repo as runtime environment, and this is written for user's who don't have the source code but want to run localstack in host mode.
i have no good suggestion of where to put this, but it's a bit out of place here. generally though, this is user doc, but we discourage it because we don't want to support host mode ;-)